GVFS/GVFS.UnitTests/Maintenance/GitMaintenanceStepTests.cs

```diff
@@ -108,5 +122,32 @@ protected override void PerformMaintenance()
                     nameof(this.SawWorkInvoked));
             }
         }
+
+        public class CheckStopDuringGitStep : GitMaintenanceStep

```

Comment:
I wonder if we should just have one class that has properties on when to call `Stop()` since these three class seem to all be the same except with if/when stop is called

Reply:
Good idea!

Reply:
[Done!](https://github.com/Microsoft/VFSForGit/pull/1072/commits/4a49bf20239919f619087ba8f104554e26b21b9e)